### ISSUE TITLE
questions about code in file "densenet_efficient.py"

### ISSUE THREAD

**Issue by zhiqiangdon** [gpleiss/efficient_densenet_pytorch#8]
Hi,

Thanks for sharing the code. I read your code and have two small questions.
1. Function "recompute_forward" in class "_EfficientBatchNorm" seems to be unused. Is it redundant?
2. "training = self.efficient_batch_norm.training" in line 266 and "self.efficient_batch_norm.training = training" in line 295 seems to be unnecessary. I think self.efficient_batch_norm.training should be true in both the first forward and the second forward when computing the gradients.

**Comment by gpleiss** [comment]
Sorry for the late reply!

1. `recompute_forward` was totally unnecessary. (Probably old code that we forgot to remove!) I'll push a change soon.

2. We turn training mode off in the second batch-norm forward to ensure that we're using the same `running_mean` and `running_var` statistics.

**Comment by wandering007** [comment]
@gpleiss @zhiqiangdon About question 2, I don't understand either. According to @gpleiss 's answer, why commenting [this line](https://github.com/gpleiss/efficient_densenet_pytorch/blob/master/models/densenet_efficient.py#L279)? aka `self.efficient_batch_norm.training = False`
